values.go

```diff
@@ -65,16 +73,26 @@ func (Optional) Type() Type {
 	return nil
 }
 
-func (o Optional) ToGoValue() interface{} {
-	if o.Value == nil {
+func (v Optional) ToGoValue() interface{} {
+	if v.Value == nil {
 		return nil
 	}
 
-	value := o.Value.ToGoValue()
+	value := v.Value.ToGoValue()
 
 	return value
 }
 
+func (v Optional) ToRuntimeValue() runtime.Value {
+	value := &interpreter.SomeValue{}
+
+	if v.Value != nil {
+		value.Value = v.Value.ToRuntimeValue().Value
+	}
+
+	return runtime.Value{Value: value}
```

Comment:
`interpreter.SomeValue{nil}` is not valid, a Cadence `nil` is represented by `interpreter.NilValue`:

```suggestion
	if v.Value == nil {
	    return runtime.Value{interpreter.NilValue{}}
	}
	
	innerValue := v.Value.ToRuntimeValue().Value
	value := interpreter.NewSomeValueOwningNonCopying(innerValue)
	return runtime.Value{Value: value}
```

Maybe also just have a `cadence.Nil` instead of a `cadence.Optional{nil}`?